### PR TITLE
fix improper determination of 'return this' status

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: test/fail_compilation/retscope.d
Comment (Geod24) [inline test/fail_compilation/retscope.d]: This is not the test case I provided. The original test case was

```D
int* escape () @safe
{
    int i;
    Foo f;
    f.v = &i;
    return f.foo;
}

struct Foo
{
    int* v;
    int* foo () @safe { return this.v; }
}
```

And still compiles with this P.R.

File: test/fail_compilation/retscope.d
Comment (Geod24) [inline test/fail_compilation/retscope.d]: I would hold off on any `scope class` test before the fix in stable gets merged into scope. @MartinNowak ?

Comment (MartinNowak) [inline test/fail_compilation/retscope.d]: I'll try to get it here.

**Comment by WalterBright** [comment]
This is very strange. The error produced is compiling Phobos unittests:
```
std\typecons.d(5374): Error: circular typeof definition
std\typecons.d(5374): Error: alias std.typecons.__unittestL5988_127.S.Proxy!(foo).ValueType recursive alias declaration
std\typecons.d(5994): Error: mixin std.typecons.__unittestL5988_127.S.Proxy!(foo) error instantiating\
```
which does not change even if I disable the changes in this PR.

**Comment by WalterBright** [comment]
https://github.com/dlang/dmd/pull/6331 seems to be the culprit.

**Comment by WalterBright** [comment]
@MartinNowak ready to rock!

File: src/escape.d
Comment (MartinNowak) [inline src/escape.d]: Coverage missing, please add a test case.
Also see https://github.com/codecov/browser-extension.

File: test/fail_compilation/retscope.d
Comment (MartinNowak) [inline test/fail_compilation/retscope.d]: It rather seems like a good idea to make tests position independent, what's the problem with it?

File: test/fail_compilation/retscope.d
Comment (MartinNowak) [inline test/fail_compilation/retscope.d]: Should be deleted now if you fixed the test.

**Review by MartinNowak — APPROVED** [review]
I'll do another rebase, things got messed up a bit.

File: src/escape.d
Comment (andralex) [inline src/escape.d]: @WalterBright you may want to install the CodeCov Chrome extension, it nicely shows uncovered lines in red on this page!

**Review by andralex — APPROVED** [review]
nice!

**Comment by MartinNowak** [comment]
Just removed the noise from the weird rebase, now it's clearer what this PR actually does.
Still please spend more than 2 seconds writing commit messages and PR descriptions, saves a lot of guesswork on the other side.
"fix improper determination of status" conveys almost no information.

**Comment by MartinNowak** [comment]
Don't try this at home kids, we're still testing our new auto-merge feature.
https://github.com/MartinNowak/dlang-bot/pull/17

**Comment by andralex** [comment]
@MartinNowak COOL COOL COOL!

**Pull request merged**
